alf/algorithms/distributed_off_policy_algorithm.py

```diff
@@ -154,16 +154,20 @@ def __init__(self,
         self._ddp_rank = max(0, PerProcessContext().ddp_rank)
         self._num_ranks = PerProcessContext().num_processes
 
-    def _opt_free_state_dict(self) -> dict:
-        """Return `self._core_alg` state dict without optimizers.
+    def _distributed_state_dict(self) -> dict:
+        """Return `self._core_alg` state dict for distributed training.
 
         This dict will be used for param syncing between a trainer and an unroller.
         Sometimes optimizers have large state vectors which we want to exclude.
+        Also we should exclude other parameters such as those of pretrained models.
         """
+        # Note that self._core_alg won't create a relay buffer so we don't have
+        # to worry about including it in the state dict.
         return {
             k: v
             for k, v in self._core_alg.state_dict().items()
-            if '_optimizers.' not in k
+            if (('_optimizers.' not in k) and (
+                not isinstance(v, torch.nn.Parameter) or v.requires_grad))
```

Comment:
can you explain why ``not isinstance(v, torch.nn.Parameter)``?

Reply:
> can you explain why `not isinstance(v, torch.nn.Parameter)`?

`requires_grad` is only available to Parameter. We should also sync buffers.